### ISSUE TITLE
TS2300: Duplicate identifier

### ISSUE THREAD

**Issue by schrobak** [graphql-nexus/nexus-plugin-prisma#537]
I have this `schema.prisma`:

```gql
model Media {
  id String @default(cuid()) @id @unique
}

model Audio {
  id String @default(cuid()) @id @unique
}
```

After generating the supporting type of nexus prisma I end up with 
```ts
interface NexusPrismaTypes {
  Query: {
    media: "Media";
    media: "Media";
    audio: "Audio";
    audio: "Audio";
  }
}
```

Any idea why those words are not changed to their equivalent plural form (event if they will be grammatically incorrect)?

**Comment by schrobak** [comment]
I actually find out that `pluralize` was causing the issue here. The solution for me was to add following line in the schema generation script:
```
import * as pluralize from "pluralize";

pluralize.addIrregularRule("media", "medias");
pluralize.addIrregularRule("audio", "audios");
```

**Comment by WellFail** [comment]
+1

**Comment by Devlin556** [comment]
+1

**Comment by denisRabov** [comment]
+1

**Comment by WellFail** [comment]
Similar problem with numbers in model names:
```
    post3: 'Post3'
    post3S: 'Post3'
``` 
Currently, last character 's' is generated in the upper case.

**Comment by jasonkuhrt** [comment]
Thanks for reporting this. Looks like we need to review our pluralize dep. it really ought to handle pretty much any case I think. Irregular shouldn’t default to doing nothing. I’m on mobile now but will check later if we can just better use the pluralize lib e.g. config.

**Comment by jonestristand** [comment]
Another example. Model name is staff, so plural is also staff. Then cannot access t.crud.staff for both the list and individual variants of the model

**Comment by wbunting** [comment]
Another thing you can do if your underlying database table name does not have a valid "pluralized" version then you can use the `@@map()` to point a model at that table and as long as the model name can be pluralized it should work as expected

**Comment by sfratini** [comment]
Also wanted to add this does not work:
```
pluralize.addIrregularRule("categories", "categoriess");
```
I am aware prisma recommends to use single instead of plural on the db model, but still, this breaks current deployment. I am on stage where I can rename tables so I am fine, but will not always be the case.

**Comment by alanstriglio** [comment]
Hello, 
since Prisma (Prisma 2) doesn't use anymore pluralize for querying Models, I think the best way to solve this is by using same methods on Prisma: 

```js
prisma.user.findOne()
prisma.user.findMany()
prisma.user.createOne()

// etc

// ... so in nexus plugin 

t.crud.user.findOne()
t.crud.user.findMany()
t.crud.user.createOne()

// etc
```

What do you think?

**Comment by bennettrogers** [comment]
I agree with @alanstriglio. Removing the need to correctly pluralize every given model name would remove a source of potential gotchas like these.
